exporter/awsemfexporter/grouped_metric.go

```diff
@@ -85,6 +97,103 @@ func addToGroupedMetric(pmd *pdata.Metric, groupedMetrics map[interface{}]*Group
 	}
 }
 
+type kubernetesObj struct{
+	Host string `json:"`
+	Labels internalLabelsObj `json:`
+	Namespace_name string `json:`
+	Pod_id string `json:`
+	Pod_name string `json:`
+	Pod_owners internalPodOwnersObj `json:`
+	Service_name string `json:`
+}
+
+type internalLabelsObj struct{
+	App string `json:`
+	Pod_template_hash string `json:`
+}
+
+type internalPodOwnersObj struct{
+	Owner_kind string `json:`
+	Owner_name string `json:`
+}
+
+
+func addKubernetesWrapper(labels map[string]string){
+	//create schema
+	schema := kubernetesObj{}
+	schema.Host = "host_name"
+	schema.Labels =
+	internalLabelsObj{
+		App: "app",
+		Pod_template_hash: "pod_template_hash",
+	}
+	schema.Namespace_name = "namespace_name"
+	schema.Pod_id = "pod_id" 
+	schema.Pod_name = "pod_name"
+	schema.Pod_owners = 
+	internalPodOwnersObj{
+			 Owner_kind: "owner_kind" , 
+			 Owner_name: "owner_name" ,
+		}
+	schema.Service_name = "service_name"
+
+	labels["kubernetes"] = recursivelyFillInStruct(labels,schema)
+}
+
+
+func recursivelyFillInStruct(labels map[string]string, schema interface{}) string{
+	jsonBytes, err := json.Marshal(schema)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Why are we not using `logger *zap.Logger`?

Reply:
Oh I just default to log, when changing it over to Zap should I retain the fact that it is a fatal log or should a failure here simply be on the error level and pushed up to PushMetricsData where it will simply skip that pusher?

Reply:
We can return the error from here and propagate through the calling functions.